x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/RestEsqlQueryAction.java

```diff
@@ -40,21 +45,64 @@ public List<Route> routes() {
 
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        // Measure response time beginning with now; also use this for logging.
+        ThreadSafeStopWatch responseTimeStopWatch = new ThreadSafeStopWatch();
+
         EsqlQueryRequest esqlRequest;
         try (XContentParser parser = request.contentOrSourceParamParser()) {
             esqlRequest = EsqlQueryRequest.fromXContent(parser);
         }
+        // Create some kind of query id so that we can correlate the beginning and the end of a query in the logs.
+        String queryId = UUIDs.base64UUID();
```

Comment:
What's the point of the id? to differentiate between queries or is it meant to be persisted somewhere?
As far as I can tell it's the former and thus needs no time component so doing queryId.hashCode() should work - it will be the same for the exact same queries but that is fine.

Reply:
The point of the ID is to be able to correlate start and end of a query, even if e.g. the same query string gets sent multiple times (user hits send twice or Kibana sends the query multiple times etc.). Mostly, I wanted to address a suggestion to log the session ID, but using something simpler because logging the session ID is actually very non-trivial.

I'll just remove it if it's not useful enough. A hash code wouldn't help since the id should be different even for the same query being sent twice.